pontoon/sync/formats/l20n.py

```diff
@@ -0,0 +1,135 @@
+                        order += 1
+
+                    for attr in obj.attrs:
+                        key = '.'.join([obj.id.name, attr.id.name])
+                        self.entities[key] = L20NEntity('attr', key, attr.value.source, '', {None:attr.value.source}, order=order)
+                        order += 1
+                    comments = []
+
+                elif obj.type == 'Comment':
+                    comments.append(obj.body)
+                else:
+                    comments = []
+
+    @property
+    def translations(self):
+        return sorted(self.entities.values(), key=lambda e: e.order)
+
+    def save(self, locale):
+        if not self.source_resource:
+            raise SyncError('Cannot save l20n resource {0}: No source resource given.'
+                            .format(self.path))
+
+        with codecs.open(self.source_resource.path, 'r', 'utf-8') as resource:
+            new_structure = L20nParser().parse(resource.read())
+
+        to_remove = []
+        for res_entity in new_structure.body:
+            if res_entity.type == 'Entity':
+                entity_id = res_entity.id.name
+                translations = self.entities[entity_id].strings
+                comments = self.entities[entity_id].comments
+
+                if not translations:
+                    entity_idx = new_structure.body.index(res_entity)
+                    for comment_entity in new_structure.body[entity_idx-len(comments):entity_idx]:
+                        to_remove.append(comment_entity)
+                    to_remove.append(res_entity)
+                    continue
+
+                if res_entity.value.type == 'String':
+                    res_entity.value.content[0] = translations[None]
+
+                elif res_entity.value.type == 'Hash':
+                    res_entity.value.items = [
+                        ast.HashItem(
+                            ast.Identifier(locale.get_cldr_plural(plural_form)),
+                            ast.String([value], value),
+                            False
+                        ) for plural_form, value in translations.items()]
+
+                for attr in res_entity.attrs:
+                    key = '.'.join([res_entity.id.name, attr.id.name])
+                    attr_translations = self.entities[key].strings
+                    if not attr_translations:
+                        res_entity.attrs.remove(attr)
+                        continue
+                    attr.value.content[0] = attr_translations[None]
+
+        for entity in to_remove:
+            new_structure.body.remove(entity)
+
+        with codecs.open(self.path, 'w+', 'utf-8') as f:
+            f.write(Serializer().serialize(new_structure))
+            log.debug("Saved file: %s", self.path)
+
+
+def parse(path, source_path=None, locale=None):
+    """Tries to parse l20n file. Raises an ParseError if file contains Junk Entries."""
```

Comment:
Do we really rease that error?

I would add the same comment we have in silme.py:
`# TODO: Cache the source resource to avoid re-parsing it a bunch.`